File: src/commands/admin-commands/syncslash.ts
import { Discord, Slash } from "discordx";
import type { CommandInteraction } from "discord.js";
import { requireAdmin } from "../../helper/adminGuard.js";
import { bot } from "../../main.js";

@Discord()
export class SyncSlash {
  @Slash({ name: "syncslash", description: "Re-synchronize all slash commands with Discord" })
  async syncSlash(interaction: CommandInteraction): Promise<void> {
    if (!requireAdmin(interaction)) return;

    await interaction.deferReply({ ephemeral: true });
    await bot.initApplicationCommands();
    await interaction.editReply({ content: "Slash commands have been re-synchronized." });
  }
}

File: unused/admin/syncslash.py
import discord
from discord.ext import commands
from utils.admin_guard import admin_check


class SyncSlash(commands.Cog):
    @discord.app_commands.command(name="syncslash", description="Re-synchronize all slash commands")
    @admin_check()
    async def sync_slash(self, interaction: discord.Interaction):
        await interaction.response.defer(ephemeral=True)
        await interaction.client.tree.sync()
        await interaction.edit_original_response(content="Slash commands synced.")


async def setup(bot):
    await bot.add_cog(SyncSlash())
